tools/clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -1038,7 +1036,14 @@ void clang::InitializePreprocessor(
   // Instruct the preprocessor to skip the preamble.
   PP.setSkipMainFilePreamble(InitOpts.PrecompiledPreambleBytes.first,
                              InitOpts.PrecompiledPreambleBytes.second);
-                          
+
+#ifdef ENABLE_SPIRV_CODEGEN
+  if (LangOpts.SPIRV) {
+    spirv::BuildPredefinesForSPIRV(Predefines,
+                                   PP.getLangOpts().EnableTemplates);
```

Comment:
TODO: When HLSL 2021 is released, we will drop `EnableTemplates`. Use HLSL version instead of it.

Reply:
Done.